app/controllers/charge_controller.js

```diff
@@ -112,98 +112,84 @@ module.exports = {
+        emailTypos = req.body['email-typo-sugestion'] !== req.body.originalemail ? commonTypos(userEmail) : null
+      }
+      try {
+        await Charge(req.headers[CORRELATION_HEADER]).patch(req.chargeId, 'replace', 'email', userEmail)
+      } catch (err) {
+        return responseRouter.response(req, res, 'SYSTEM_ERROR', withAnalytics(charge))
+      }
+    }
 
-          if (data.validation.hasError || emailTypos) {
-            if (emailTypos) {
-              data.validation.hasError = true
-              data.validation.errorFields.push({
-                cssKey: 'email-typo',
-                value: i18n.__('fieldErrors.fields.email.typo')
-              })
-              data.validation.typos = emailTypos
-              data.validation.originalEmail = userEmail
-            }
-            charge.countries = countries
-            return appendChargeForNewView(charge, req, charge.id).then(
-              () => {
-                _.merge(data.validation, withAnalytics(charge, charge), _.pick(req.body, preserveProperties))
-                responseRouter.response(req, res, views.CHARGE_VIEW, data.validation)
-              },
-              err => {
-                logging.failedGetWorldpayDdcJwt(err)
-                responseRouter.response(req, res, 'SYSTEM_ERROR', withAnalytics(charge))
-              }
-            )
-          }
-          AWSXRay.captureAsyncFunc('Charge_email_patch', function (subSegment) {
-            emailPatch
-              .then(() => {
-                subSegment.close()
-                const correlationId = req.headers[CORRELATION_HEADER] || ''
-                const payload = normalise.apiPayload(req, card)
-                if (res.locals.service.collectBillingAddress === false) {
-                  delete payload.address
-                }
-                connectorClient({ correlationId }).chargeAuth({ chargeId: req.chargeId, payload })
-                  .then(handleCreateResponse(req, res, charge))
-              })
-              .catch(err => {
-                subSegment.close(err.message)
-                logging.failedChargePatch(err.message)
-                responseRouter.response(req, res, 'ERROR', withAnalyticsError())
-              })
-          }, clsSegment)
+    if (data.validation.hasError || emailTypos) {
+      if (emailTypos) {
+        data.validation.hasError = true
+        data.validation.errorFields.push({
+          cssKey: 'email-typo',
+          value: i18n.__('fieldErrors.fields.email.typo')
         })
-    }, clsSegment)
+        data.validation.typos = emailTypos
+        data.validation.originalEmail = userEmail
+      }
+      charge.countries = countries
+      try {
+        await appendChargeForNewView(charge, req, charge.id)
+      } catch (err) {
+        logging.failedGetWorldpayDdcJwt(err)
+        return responseRouter.response(req, res, 'SYSTEM_ERROR', withAnalytics(charge))
```

Comment:
I'm uncomfortable with the number of places that this method can terminate, there are ~5 and it makes it almost impossible to know how many combinations of things can happen. 

I know this isn't part of this PR but a much easier to reason about structure here would be to run validation, come to a conclusion and then only respond once - either with success or failure. Responding in-line makes it hard to know if this will do exactly what the previous code did.

Reply:
i agree... i might do more refactoring to reduce the termination places but the 
```
        logging.failedGetWorldpayDdcJwt(err)
        return responseRouter.response(req, res, 'SYSTEM_ERROR', withAnalytics(charge)
```
was there before.